src/library/f3d_engine.cxx

```diff
@@ -59,6 +59,12 @@ engine::~engine()
   delete this->Internals;
 }
 
+//----------------------------------------------------------------------------
+void engine::setOptions(options& opt)
```

Comment:
Sorry I don't like a `setOptions` function making the input invalid. In this case, I would prefer to explicitly call `engine.getOptions() = std::move(myOptions);`.  
Alternatively, you can have these two functions:
```cpp
void engine::setOptions(const options& opt); // call copy
void engine::setOptions(options&& opt); // call move
```
Having both will allow the user to call `engine.setOptions(std::move(myOptions))` if he wants to move it.